docs/interfaces/pyapi/pyapi.rst

```diff
@@ -0,0 +1,425 @@
+.. highlight:: none
+.. _sec-interfaces-pyapi:
+
+#########################
+Python Interface - ctypes
+#########################
+
+ctypes is a foreign function library for Python. It provides C compatible data
+types, and allows calling functions in DLLs or shared libraries. It can be used
+to wrap these libraries in pure Python (cf. `ctypes documentation 
+<https://docs.python.org/3/library/ctypes.html>`_).
+
+Before going through the following sections, please make sure that you have
+installed a working version of the ctypes package, using your preferred
+installation procedure. Also, consider adding the Python file
+`tools/pythonapi/dftbplus.py`, which contains the actual ctypes interface,
+to your PYTHONPATH environment variable or symlink it. The scripts below assume
+that the interface can be imported by invoking ``import dftbplus``.
+
+Regarding the **units**: Just like DFTB+, the interface expects and delivers
+values in **atomic units**!
+
+.. note::
+
+    To instruct DFTB+ that a dynamically linked shared library should be
+    created, the `WITH_API` and `BUILD_SHARED_LIBS` flag in the cmake
+    configuration file `config.cmake` must be set to TRUE, before starting
+    the compilation process!
+    If you don't want to mess around with files, a construct like the
+    following is a convenient way to specify these flags, while configuring
+    cmake:
+    ``cmake -DBUILD_SHARED_LIBS=1 -DWITH_API=1 -DCMAKE_TOOLCHAIN_FILE=../sys/gnu.cmake ..``
```

Comment:
At this point it would be helpful to explain the build *and* installation of DFTB+ and where the `libdftbplus.*` is found in which particular step. Most users probably don't know about the internals of the CMake build.

Reply:
Along with the changes due to the installation with setup.py, two additional comments are probably useful:

- a reference to the INSTALL.rst file (at the current state of DFTB+, the user can hardly avoid dealing with CMake, at least to a certain extent)
Here it is explained how DFTB+ has to be built for this special usecase.
- a short explanation of what the user can expect as the location of _libdftbplus.*_
(with and without 'make install')